components/constellation/constellation.rs

```diff
@@ -750,6 +771,98 @@ where
         namespace_id
     }
 
+    fn get_event_loop(
+        &mut self,
+        host: &Host,
+        top_level_browsing_context_id: &TopLevelBrowsingContextId,
+        opener: &Option<BrowsingContextId>,
+    ) -> Option<Weak<EventLoop>> {
```

Comment:
If this returned a `Result` then you could use `?` notation.

Reply:
I don't know, I added a few `warn` statements in the `None` cases...

Reply:
Well you  could make the return type `Result<Weak<EventLoop>,&'static str>` and then replace the `match foo { ... }` by `foo.ok_or("a warning message")?` Then put the logic to `warn!` in the caller of `get_event_loop`. Not a huge deal, just might make the code a bit shorter and easier to read.

Reply:
? works with Option as well, unless I'm misunderstanding the suggestion here.

Reply:
True, but then you couldn't return a warning message.

Reply:
Ok, I'll look into it, sorry I missed those responses earlier...